### DISCUSSION TITLE
Score Reranker udf?

### DISCUSSION THREAD

**Discussion by JackTan25** [milvus-io/milvus#33252] [Q&A and General discussion]
Hello, I'm trying to use milvus's multi-column top-k result. But I find the recall is very low. Finally, I find the reason lies that milvus use https://milvus.io/docs/reranking.md#Weighted-Scoring-WeightedRanker, which will not use distance as score. I hope we can use the origin distance as the score computation, that's `(w0 * L2(vector0, v_const0) + w1 * L2(vector1,v_const1) + w2 * L2(vector2,v_const2))` by using WeightRanker(w0,w1,w2). Hope add new ranker to implement this one.

**Comment by yhmo** [comment]
Wait @czs007 to comment.

**Reply by czs007** [reply]
@JackTan25  please refer to this code. https://github.com/milvus-io/milvus/blob/648d5661ca8771fadc664f427ac330b083b8734e/internal/proxy/reScorer.go#L84

Milvus currently does not support passing UDF (User-Defined Function) during ranking.
The quickest way to make modifications is to add a parameter to the WeightRanker and switch back to the previous calculation method.
If you are in a hurry, you can directly modify the code here.

**Comment by JackTan25** [comment]
thanks @czs007  I will modify the source code and test it locally

**Comment by JackTan25** [comment]
<img width="1163" alt="image" src="https://github.com/milvus-io/milvus/assets/60096118/59c083c1-bd5a-4d74-9d5b-b6bd025339ec">
@czs007 I modify only one row code, but after test, I find out the recall is zero. That's very strange. Maybe my modification is wrong??

**Reply by JackTan25** [reply]
bug here: https://github.com/milvus-io/milvus/issues/33294
